### PR TITLE
Enable Fine-Tuning on AMPLIFY model embeddings 

### PR DESCRIPTION
**Pull request by philippschw**
A typical use case is to use the embeddings as a base model and fine-tune it on dataset specific tasks. A popular approach is for example: [Light Attention](https://github.com/HannesStark/protein-localization/blob/master/models/light_attention.py)
This requires a few small modifications to the embeddings.py file.

- Normalize the embeddings using L2 normalzation (this step is the most critical!)
- Convert embeddings to HalfTensor for reduced memory and storage space requirements
- Do not convert to numpy as this operation is very slow

### CODE REVIEW

**Comment by qfournier** [comment]
Thank you for your suggestions. I hadn't considered normalizing the residue embeddings! I quickly tested it on a downstream task, and so far, it seems to yield comparable results.

The `Embedder` class was not intended to be used for fine-tuning AMPLIFY. Instead, I suggest you create a dedicated classifier class with the pre-trained model as the trunk and feedforward layers on top for classification.

```python
class amplify_classifier(nn.Module):
    def __init__(self, pretrained_model_name_or_path, hidden_size, num_labels):
        super().__init__()
        self.trunk = AutoModel.from_pretrained(pretrained_model_name_or_path, trust_remote_code=True)
        self.classifier = nn.Sequential(
            nn.Linear(self.trunk.config.hidden_size, hidden_size),
            nn.ReLU(),
            nn.Linear(hidden_size, num_labels),
        )

    def forward(self, input_ids, attention_mask, frozen_trunk=True, normalize_hidden_states=True):
        with torch.no_grad() if frozen_trunk else torch.enable_grad():
            h = self.trunk(input_ids, attention_mask, output_hidden_states=True).hidden_states[-1]
        if normalize_hidden_states:
            h = torch.nn.functional.normalize(h, p=2, dim=-1)
        return self.classifier(h)
```

Otherwise, I strongly suggest using mixed-precision as this is generally more stable than forcing all computations to `fp16`. You can enable mixed-precision using PyTorch's `autocast` context manager.

```python
with torch.autocast(device_type="cuda", dtype=torch.float16):
    # Your training loop or forward pass
    outputs = model(input_ids, attention_mask)
    loss = loss_fn(outputs, labels)
```
